libs/sdk-backend-tiger/src/backend/workspaces/index.ts

```diff
@@ -91,9 +95,13 @@ class TigerWorkspaceQuery implements IWorkspacesQuery {
         search?: string,
     ): Promise<IWorkspacesQueryResult> {
         const allWorkspaces = await this.authCall((client) => {
+            // Should get all workspaces when searching
```

Comment:
As we are changing the Tiger backend library, it affects TIGER/PANTHER only. Rights? So, I think on TIGER/PANTHER when the search is not empty, we will search on all workspaces not only the children of a workspace.

Reply:
It's not about affecting Tiger/Panther or not, it's about the improper of putting app-specific logic in the SDK.
＊ `"Searching for all workspaces when the search param is not empty"` is just a requirement of the feature we are currently working on, the SDK should not be a place for us to implement such logic.

The SDK can be used by multiple apps and the logic in each app might be very different from each other, which means, it might be a case when GoodData (or some other companies) in the future implements a different app with a different requirement like `"Searching for all children of the current workspace when the search param is not empty"`. In such case, how can the app be implemented?

Reply:
I think if they want different logic, they need to clone to another backend library like the Lion backend :) As Bear and Tiger are different logics now. Or do you have other suggestions to achieve this requirement "The search will search on all workspaces"?

Reply:
I have updated this. Thanks!